core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java

```diff
@@ -93,12 +95,26 @@ public void evaluateResults(long lineCount, long errorCount) {
                     .$(']').$();
         }
 
-        // make up field names if there is no header
         for (int i = 0; i < fieldCount; i++) {
             if (!header || columnNames.getQuick(i).length() == 0) {
                 tempSink.clear();
                 tempSink.put('f').put(i);
-                columnNames.setQuick(i, tempSink.toString());
+                String colName = tempSink.toString();
+
+                for (int attempt = 0; attempt < 20; attempt++) {
+                    if (!uniqueColumnNames.contains(colName)) {
+                        break;
+                    }
+
+                    tempSink.put('_');
+                    colName = tempSink.toString();
```

Comment:
you can move this line out of the loop, and test `contains` using the sink

Reply:
Fixed